src/client/hud.cpp

```diff
@@ -862,6 +862,58 @@ void Hud::drawSelectionMesh()
 	}
 }
 
+void Hud::toggleBlockBounds()
+{
+	m_block_bounds_mode++;
+
+	if (m_block_bounds_mode > 2) {
+		m_block_bounds_mode = 0;
+	}
+}
+
+void Hud::drawBlockBounds()
+{
+	if (!m_block_bounds_mode) {
+		return;	
+	}
+
+	video::SMaterial old_material = driver->getMaterial2D();
+	driver->setMaterial(m_selection_material);
+
+	auto pos = player->getStandingNodePos();
+
+	auto blockPos = v3s16(
+		floorf((float) pos.X / 16.0f),
+		floorf((float) pos.Y / 16.0f),
+		floorf((float) pos.Z / 16.0f)
+	);
+
+	auto offset = intToFloat(client->getCamera()->getOffset(), BS);
+
+	s8 radius = 2;
+
+	auto halfNode = v3f(BS, BS, BS) / 2.0f;
+
+	for (s8 x = -radius; x <= radius; x++) {
+		for (s8 y = -radius; y <= radius; y++) {
+			for (s8 z = -radius; z <= radius; z++) {
```

Comment:
If wanted, you may nest them as follows to reduce indent levels:
```suggestion
	for (s8 x = -radius; x <= radius; x++)
	for (s8 y = -radius; y <= radius; y++)
	for (s8 z = -radius; z <= radius; z++) {
```

(attention for closing brackets)